Kitodo-API/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java

```diff
@@ -52,20 +74,242 @@ public KitodoServiceLoader(Class clazz, String modulePath) {
+                        String filePath = session.getServletContext().getRealPath(File.separator + PAGES_FOLDER)
+                                + File.separator + moduleName;
+                        FileUtils.deleteDirectory(new File(filePath));
+                        String resourceFolder = String.join(File.separator,
+                                Arrays.asList(temporaryFolderName, META_INF_FOLDER, RESOURCES_FOLDER));
+                        copyFrontEndFiles(resourceFolder, filePath);
+                        FileUtils.deleteDirectory(tempFolderFile);
+                    } else {
+                        logger.info("No module found in JarFile '" + jarFile.getName() + "'.");
+                    }
+                }
+            }
+        }
+        catch (Exception e) {
+            logger.error("Classpath could not be accessed", e.getMessage());
+        }
+    }
+
+    /**
+     * Extracts the module name of the current module by finding the pom.properties
+     * in the folder given by the temporary folder name
+     *
+     * @param temporaryFolderName
+     *            folder name in which the pom.properties file will be searched for
+     *
+     * @return boolean
+     */
+    private String extractModuleName(String temporaryFolderName) throws IOException {
+        String moduleName = "";
+        File properties = findFile(POM_PROPERTIES_FILE, temporaryFolderName);
+        try (InputStream input = new FileInputStream(properties)) {
+            Properties prop = new Properties();
+            prop.load(input);
+            moduleName = prop.getProperty(ARTIFACT_ID_PROPERTY);
+        } catch (FileNotFoundException e) {
+            logger.error(e.getMessage());
+        }
+        return moduleName;
+    }
+
+    /**
+     * Copies extracted frontend files by a given source folder name to the destination Folder
+     * given by a destination folder name
+     *
+     * @param sourceFolder
+     *            copies all extracted frontend files
+     * @param destinationFolder
+     *            jarFile that will be checked for frontend files
+     */
+    private void copyFrontEndFiles(String sourceFolder, String destinationFolder) throws IOException {
+        FileUtils.copyDirectory(new File(sourceFolder), new File(destinationFolder));
+    }
+
+    /**
+     * Checks, whether a passed jarFile has frontend files or not. Returns true, when the jar contains
+     * a folder with the name "resources"
+     *
+     * @param jarPath
+     *            jarFile that will be checked for frontend files
+     * @param destinationPath
+     *            destination path, where the frontend files will be extracted to
+     *
+     */
+    private void extractFrontEndFiles(String jarPath, String destinationPath) throws IOException {
+        File destinationFolder = new File(destinationPath);
+        if (!destinationFolder.exists()) {
+            destinationFolder.mkdir();
+        }
+
+        JarFile jar = new JarFile(jarPath);
```

Comment:
Class `DirectoryStream` implements the `Closeable` / `Autocloseable` interface which means that we can use the try-with-resources-statement here and the later jar.close() could be removed.